spec/services/tag_search_service_spec.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe TagSearchService do
+  describe '#call' do
+    let!(:one) { Fabricate(:tag, name: 'one') }
+
+    before { Fabricate(:tag, name: 'two') }
+
+    it 'runs a search for tags' do
+      results = subject.call('#one', limit: 5)
```

Comment:
One thing I noticed here in the service class - the `@limit` value is set within the class and calls `to_i`, so when nil it gets set to zero, and so needs to be passed in explicitly here. Practically this is fine because the only place it's called in the codebase does pass in a value.